contracts/src/PDPService.sol

```diff
@@ -4,15 +4,15 @@ pragma solidity ^0.8.13;
 import {BitOps} from "../src/BitOps.sol";
 import {Cids} from "../src/Cids.sol";
 import {MerkleVerify} from "../src/Proofs.sol";
-import {PDPRecordKeeper} from "../src/PDPRecordKeeper.sol";
+import {PDPApplication} from "../src/PDPRecordKeeperApplication.sol";
```

Comment:
I would try to avoid this circular dependency. I think you can do so by moving the operation type and so on into this contract. It makes sense to me that applications depend on details about how this contract works but not the other way around.